Common++/header/IpAddress.h

```diff
@@ -7,6 +7,17 @@
 #include <algorithm>
 #include <ostream>
 
+#ifndef PCPP_DEPRECATED
+#if defined(__GNUC__) || defined(__clang__)
+#define PCPP_DEPRECATED __attribute__((deprecated))
+#elif defined(_MSC_VER)
+#define PCPP_DEPRECATED __declspec(deprecated)
+#else
+#pragma message("WARNING: DEPRECATED feature is not implemented for this compiler")
```

Comment:
maybe #error else we will just ignore this message and never catch it in the CI no ?

Reply:
I didn't mark this as an error because I don't want to break the API. If someone uses this API they should get a warning and hopefully switch to the new API. In later versions we can remove it altogether

Reply:
I was more thinking PCAPP_DEPRECATED not defined is an error.

Reply:
I don't think it should happen that `PCAPP_DEPRECATED`, but just in case - we define it and output this warning...